Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -230,6 +231,20 @@ CookieStore::~CookieStore()
     m_mainThreadBridge->detach();
 }
 
+URL CookieStore::creationURL(ScriptExecutionContext& context) const
+{
+    auto url = context.creationURL();
+    if (context.isDocument()) {
+        RefPtr document = dynamicDowncast<Document>(context);
+        RefPtr parentDocument = document->protectedParentDocument();
+        while (SecurityPolicy::shouldInheritSecurityOriginFromOwner(url)) {
+            url = parentDocument->creationURL();
```

Comment:
What if parentDocument is null?

Reply:
Good point--we should throw an error if we can't find a creation URL. Just added this in the new iteration.

Reply:
As suggested by Sihui, I changed it to use Document::cookieURL(), so now we don't access the parentDocument anymore.